### PR TITLE
feat: add ability to launch targeted dskit modules in the grafana server CLI command

### PR DESCRIPTION
**Pull request by mildwonkey**
This commit adds a ModuleServer and ModuleRunner suitable for launching dskit services and updates the server cli command to use this instead of the full Server. The default behavior is unchanged and will launch the full Grafana server. Individual services are targeted by setting target=comma,seperated,list in the config file.

TODO
- [x]  require dev mode to start dskit modules 
- [x]  enterprise PR: https://github.com/grafana/grafana-enterprise/pull/5707

I recommend y'all try it out yourselves, but here's the output if I set a `target` in my config file but run the CLI in anything other than dev mode:
Screenshot 2023-08-31 at 10 26 29 AM

### CODE REVIEW

File: pkg/server/wire.go [hunk 1]
```diff
@@ -325,7 +328,6 @@ var wireBasicSet = wire.NewSet(
 	grpcserver.ProvideHealthService,
 	grpcserver.ProvideReflectionService,
 	interceptors.ProvideAuthenticator,
-	setting.NewCfgFromArgs,
```
Comment (mildwonkey) [inline pkg/server/wire.go:328]: This "dependency" only exists so wire can handle calling that method on the setting.CommandLineArgs, so I refactored the various Initialize commands to take a `setting.Cfg` and made sure we were calling `setting.NewCfgFromArgs` explicitly where needed. 

This does not change the setting.Provider (oss & enterprise both); that is a more idiomatic dependency.

It's an unexpected, free baby step towards refactoring setting.Cfg!!

**Comment by mildwonkey** [comment]
I'm going to follow up, but the deploy-to-hg step failed because it checked out the enterprise branch of grafana, not my PR:

```
2023/08/31 18:37:29 Cloning Grafana Enterprise...
2023/08/31 18:37:29 if git -C src checkout main; then echo "checked out main"; else exit 3; fi
```

File: pkg/server/module_server.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/grafana/pkg/api"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/modules"
+	grafanaapiserver "github.com/grafana/grafana/pkg/services/grafana-apiserver"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// NewModule returns an instances of a ModuleServer, responsible for managing
```
Comment (zserge) [inline pkg/server/module_server.go:22]:
```suggestion
// NewModule returns an instance of a ModuleServer, responsible for managing
```

**Review by zserge — APPROVED** [review]

**Pull request merged**